serializer/serializer.c

```diff
@@ -222,6 +222,29 @@ status_t iota_transaction_to_json_object(iota_transaction_t const* const txn, cJ
   return SC_OK;
 }
 
+static status_t transaction_array_to_json_array(cJSON* json_root, char* obj_name,
+                                                const transaction_array_t* const txn_array) {
+  status_t ret = SC_OK;
+  iota_transaction_t* txn = NULL;
+  cJSON* txn_obj = NULL;
+  cJSON* array_obj = cJSON_CreateArray();
+
+  if (array_obj == NULL) {
+    return SC_SERIALIZER_JSON_CREATE;
+  }
+  cJSON_AddItemToObject(json_root, obj_name, array_obj);
+
+  TX_OBJS_FOREACH(txn_array, txn) {
+    txn_obj = NULL;
+    ret = iota_transaction_to_json_object(txn, &txn_obj);
+    if (ret != SC_OK) {
+      return ret;
```

Comment:
`array_obj` should be freed here.

Reply:
I check out entangled they didn't free JSON array object.

Reply:
Wouldn't that cause memory leak? :thinking:

Reply:
The address sanitizer didn't return any error, so I think it's fine 0.0